### DISCUSSION TITLE
Phase A Voltage low - tips for calibration?

### DISCUSSION THREAD

**Discussion by eklee** [emporia-vue-local/esphome#21] [Q&A] [answered]
My voltage for Phase A seems kind of low while Phase B looks high:
![image](https://user-images.githubusercontent.com/1013005/187305222-4ef3d8b1-4cd8-454b-926b-8ca42dd5c3ef.png)

What is the best way to calibrate the voltages? I am thinking of using a multimeter to measure the voltage between the Vue's black breaker and ground to get phase A voltage, and Vue's red breaker and ground to get phase B voltage. Then use the formula `<in-use calibration value> * <accurate voltage> / <reporting voltage>` to derive the `calibration` value in the yaml file. Is this the right approach?

BTW, thanks for the great project!

**Comment by flaviut** [comment, marked as answer]
> `<in-use calibration value, default .022> * <accurate voltage> / <reporting voltage>`

Spot-on!

It would be nice to automatically do this, since the calibration values depend on which order the phases are hooked up, but I haven't been able to reverse engineer this part of things. Ghirda doesn't disassemble floats well :(

**Comment by Jens5395** [comment]
I went through recalibrating and came up with  calibration: 0.02292  and 0.0214 for phase A and B respectively. I used a trial and error approach of tweaking the number, uploading the new configuration, check the voltage shown and repeat until I was close enough. While it sounds tedious, it wasn't really that bad.
